### PR TITLE
Animate splash screen

### PR DESCRIPTION
**Pull request by zoontek**
### Details

This PR adds support for animated splash screen , instead of a simple fade out.

A few things:

- The bootsplash module has been updated to handle promises, in the exact same way as [react-native-bootsplash v4](https://github.com/zoontek/react-native-bootsplash). This is important to wait that the native splash screen is hidden before firing the animation.
- This PR does not rely on `react-native-reanimated`. As it's only scale and opacity, it's already 100% native animations (I also tried with reanimated, the behaviour / smoothness were identical)
- On the web, I kept a simple fade as it's not a common pattern to have an animated splash screen on this platform (same I tried it, it felt weird 😅)
- I removed the `fallback` prop on `NavigationContainer`, which was responsible for this ugly loading screen just after the splash screen hide. It appears their documentation say: "If you have a native splash screen, please use [onReady](https://reactnavigation.org/docs/navigation-container/#onready) instead of fallback prop.", so this is the way to go ([Link](https://reactnavigation.org/docs/navigation-container/#fallback))
- I set the status bar to `BaseAppTheme` `android:statusBarColor` on Android, at start. Before, it was the color of the splash screen: it worked since we faded the native splash screen as a whole, making the system bars fade with it. It doesn't work with this new approach since the system bars cannot just switch from splash screen background color to app status bar color when removing the native splash screen (without fade). So instead, I immediately apply the app status bar color. It continues to "blink" in development on Android >= 12 (system thread splash screen uses transparent system bars, app thread splash screen does not), but it's not an issue on release as it's nearly immediate (check videos)
- I added a `navigationBarHeight` constant to be able to center the logo on window height (including system bars). It uses a logic 100% identical to [react-native StatusBar module](https://github.com/facebook/react-native/blob/v0.72.0-rc.1/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/statusbar/StatusBarModule.java#L51).
- I re-generated the assets since they were 1px off compared to the SVG (use Github Swipe view to see the small difference)

### Fixed Issues

$ https://github.com/Expensify/App/issues/14151
PROPOSAL: https://github.com/Expensify/App/issues/14151#issuecomment-1510516375

$ https://github.com/Expensify/App/issues/18172

### Tests

1. Compiles and starts the app on a device with iOS or Android
2. Verify that the splash screen is animated and dismissed
3. Check that there is no intermediate spinner between the splash screen an the login form (like in [those videos](https://github.com/Expensify/App/issues/18172#issue-1689136798))

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

No changes (still a simple fade):

https://user-images.githubusercontent.com/1902323/232309117-f29193e3-7fbc-4c35-9007-6b7e93ac4854.mp4

</details>

<details>
<summary>Mobile Web - Chrome</summary>

No changes, similar to Web version.

</details>

<details>
<summary>Mobile Web - Safari</summary>

No changes, similar to Web version.

</details>

<details>
<summary>Desktop</summary>

No changes, similar to Web version.

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/1902323/232308861-aaade986-e525-40a2-af02-99c997cd27ba.mp4

</details>

<details>
<summary>Android</summary>

Android 13 (release, Pixel 4a, high-end device):

https://user-images.githubusercontent.com/1902323/232309509-38e1be71-5dd0-4646-9a0a-94ff42fbf15b.mp4

---

Android 10 (release, Xiaomi A2, low-end device):

https://user-images.githubusercontent.com/1902323/232309564-2c47b2ab-7146-4c8f-9c66-b8ced28eee0f.mp4

</details>

### CODE REVIEW

**Comment by Gonals** [comment]
rerolling pullerbear, as I'm OOO on parental leave

**Comment by marcochavezf** [comment]
Hi @zoontek! Could you fix the conflict?

**Comment by zoontek** [comment]
@marcochavezf Done

**Comment by zoontek** [comment]
@0xmiroslav Any news?

**Review by 0xmiroslav — COMMENTED** [review]
Android animation doesn't work at all for me. Tested on Android 13
NOTE: I merged main into this branch locally before testing

https://user-images.githubusercontent.com/97473779/234852990-8eadc094-bb3f-4ed5-a580-6059ac8aa6e4.mov

File: src/Expensify.js
Comment (zoontek) [reply src/Expensify.js:95]: That would result in huge code duplication.
I will merge main

Comment (zoontek) [reply src/Expensify.js]: Anyway, I found a way to remove them by making `AnimatedSplashScreen` managing its own state.

**Comment by marcochavezf** [comment]
> @marcochavezf can you please add `Ready To Build` label, so we can test release build?

Done

**Comment by zoontek** [comment]
@0xmiroslav I fixed the issue! It appears that the bootsplash module can be faster to hide the native splash screen than `Animated.View` to render.

So I updated the logic, renamed `AnimatedSplashScreen` to `SplashScreenHider`, which is responsible to hide the native splash screen and animate the JS one when `onLayout` is triggered (instead of using `useEffect`)

I restarted the app a dozen of times, I cannot reproduce it anymore 🎉
So it's ready for a second review.

**Comment by marcochavezf** [comment]
Oh conflict again, @zoontek could you resolve it?

**Comment by zoontek** [comment]
@marcochavezf Done ✅

**Comment by marcochavezf** [comment]
Cool thanks! Re-run [action](https://github.com/Expensify/App/actions/runs/4874434368) to build and deploy apps.

**Comment by zoontek** [comment]
@0xmiroslav I clearly see it in the video. This **has** to be fast since the app is already loaded and it's just additional wait for the user. It's approximately the same duration as the previous fade.

**Comment by zoontek** [comment]
@0xmiroslav I'm on it.
PS: Given the fact that I also fixed https://github.com/Expensify/App/issues/18172 for free here can I get the Upwork job too?

This was caused by:

> I removed the fallback prop on NavigationContainer, which was responsible for this ugly loading screen just after the splash screen hide. It appears their documentation say: "If you have a native splash screen, please use [onReady](https://reactnavigation.org/docs/navigation-container/#onready) instead of fallback prop.", so this is the way to go ([Link](https://reactnavigation.org/docs/navigation-container/#fallback))

File: src/components/SplashScreenHider/index.js
Comment (zoontek) [reply src/components/SplashScreenHider/index.js:63]: Yes, it's only for android.
I switched too:

```ts
marginTop: -(StatusBar.currentHeight || 0),
marginBottom: -(BootSplash.navigationBarHeight || 0),
```

**Comment by zoontek** [comment]
@0xmiroslav All done ✅

**Comment by zoontek** [comment]
I updated the description to include https://github.com/Expensify/App/issues/18172, as it has been assigned to me 🙂

**Comment by zoontek** [comment]
@0xmiroslav I just tried and I cannot reproduce it (even by throttling network to slow 3G):

https://github.com/Expensify/App/assets/1902323/764083e8-4856-4481-b8ce-c1ac92efdc92

https://github.com/Expensify/App/assets/1902323/47000455-8981-46cf-809e-62f7f34de710

I rebased against main ✅

**Comment by zoontek** [comment]
@0xmiroslav I see an error in your console (refusing to execute the inline script because of CSPs), is the splash(.js) screen script simply blocked on your test domain?

**Comment by zoontek** [comment]
@0xmiroslav Yeah, I can reproduce it too: the splash screen is not even shown on web because of your [CSP](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP):

```
17477.pr-testing.expensify.com/:59  Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'self' https://polyfill.io https://cdn.plaid.com https://www.woopra.com https://assets.onfido.com https://sentry.io". Either the 'unsafe-inline' keyword, a hash ('sha256-EKld/TJJraWN2+wMTu3RaTz3weV2JN/KDL7GMWZ78qQ='), or a nonce ('nonce-...') is required to enable inline execution.
```

You have to add `unsafe-inline` policy for `expensify.com *.expensify.com` 🙂

But the PR works correctly.

**Comment by zoontek** [comment]
@0xmiroslav No, what I mean is that there a content security policy issue on your server here preventing the splash screen to been shown.

By using [CSP unblock](https://chrome.google.com/webstore/detail/csp-unblock/lkbelpgpclajeekijigjffllhigbhobd) you can clearly saw it:

https://github.com/Expensify/App/assets/1902323/f0cca9e2-32bf-417f-b943-33965d31e391

The PR works correctly, your CSP config just need some tweaking here 😄
Forward that to someone who's able to update them on the pr-testing server

**Comment by zoontek** [comment]
It appears the issue also exists on your deployed production. I spotted a bug:

Screenshot 2023-05-10 at 11 21 40

**Comment by zoontek** [comment]
@0xmiroslav uh, I think this is not a decision to make, you have to fix your CSPs to make the splash screen work, that's all.

**Comment by zoontek** [comment]
@0xmiroslav Hi again! I submitted to @roryabraham a solution to fix this CSP situation and pushed it.

I removed the inlined script and now hide the splash screen when `<SplashScreenHider/>` is mounted, which simplify a lot the logic (no need for the setInterval + the minimum amount of time to wait anymore 🎉)

**Comment by zoontek** [comment]
@0xmiroslav No worry, I rebased it ✅

File: src/components/Pressable/GenericPressable/BaseGenericPressable.js
Comment (zoontek) [inline src/components/Pressable/GenericPressable/BaseGenericPressable.js:66]: I just ran `npm run prettier`, this file wasn't formatted

**Comment by zoontek** [comment]
I fixed the `getVisibilityStatus` one (not a huge deal, could have been easily catched with TypeScript since it's a difference in the module contracts between native and web).

I rebased against main but now I can't test the app because of https://github.com/Expensify/App/issues/18653.
Should I fix that too?

**Comment by zoontek** [comment]
@0xmiroslav I guess you can do another review. The `hide` looks like a the native module is not linked anymore (maybe a regression pushed on main? - because it was working when this PR was opened)

**Comment by zoontek** [comment]
@0xmiroslav I cannot reproduce the issue on iOS. Just replaced `RCT_REMAP_METHOD` for `RCT_EXPORT_METHOD` for future new arch.

Should be ready to review. Please do it as soon as possible.

**Comment by zoontek** [comment]
> We should be careful while this renaming. Please also consider desktop app.

I did the opposite: rename `SplashScreenHider` to .native (even if there's no issue in development mode)

**Comment by zoontek** [comment]
> > I rebased against main but now I can't test the app because of #18653.
> 
> This is iOS issue. you can still test android app.

Tested both (#18653 is easily fixable with a Podfile workaround), both works correctly.

**Comment by zoontek** [comment]
@0xmiroslav Yeah, this is because of the `Platform` removal.
Handling state in the parent component made sense after all 😄

**Comment by zoontek** [comment]
I will re-introduce `Platform` and upper component state to prevent this kind of issues. And put a comment on why it's better do it that way.

**Comment by zoontek** [comment]
@0xmiroslav Fixed it. Actually, there is no need to use `Platform` since this PR also include the fix for the CSP issue, and at start if `SplashScreenHider` was a noop on web, it's not the case anymore since [this commit](https://github.com/Expensify/App/pull/17477/commits/7e2f8a7c07a4a83733d771ec8afce25e6b307622)

**Review by 0xmiroslav — DISMISSED** [review]
Looks good, tests well in dev.
@roryabraham or @marcochavezf can you please trigger build again so I can test and confirm no issue in release build?

**Comment by zoontek** [comment]
@0xmiroslav ✅ Done. Note to myself for the next time: add new style in the middle of `styles.js` to avoid conflicts 😅

**Review by 0xmiroslav — DISMISSED** [review]
Release builds also test well.
@marcochavezf all yours!

**Review by marcochavezf — DISMISSED** [review]
LGTM, thank guys! @roryabraham would you want to give it a final review?

**Comment by zoontek** [comment]
@0xmiroslav @marcochavezf @roryabraham Done (with a merge commit, not a rebase)

**Review by 0xmiroslav — APPROVED** [review]

**Comment by zoontek** [comment]
@0xmiroslav Done ✅

**Review by 0xmiroslav — DISMISSED** [review]

File: src/components/SplashScreenHider/index.native.js
Comment (roryabraham) [inline src/components/SplashScreenHider/index.native.js:1]: Can we plz update this to use Reanimated instead of React Native's Animated lib? I want to try and standardize on Reanimated as much as possible across the app.

```diff
diff --git a/src/components/SplashScreenHider/index.native.js b/src/components/SplashScreenHider/index.native.js
index a32cdb8169..8e544b7da2 100644
--- a/src/components/SplashScreenHider/index.native.js
+++ b/src/components/SplashScreenHider/index.native.js
@@ -1,6 +1,7 @@
-import {useCallback, useRef, useState} from 'react';
+import {useCallback, useRef} from 'react';
 import PropTypes from 'prop-types';
-import {Animated, Easing, StatusBar, StyleSheet} from 'react-native';
+import {StatusBar, StyleSheet} from 'react-native';
+import Reanimated, {useSharedValue, withTiming, Easing, useAnimatedStyle, runOnJS} from 'react-native-reanimated';
 import BootSplash from '../../libs/BootSplash';
 import Logo from '../../../assets/images/new-expensify-dark.svg';
 import styles from '../../styles/styles';
@@ -17,8 +18,16 @@ const defaultProps = {
 const SplashScreenHider = (props) => {
     const {onHide} = props;
 
-    const [opacity] = useState(() => new Animated.Value(1));
-    const [scale] = useState(() => new Animated.Value(1));
+    const opacity = useSharedValue(1);
+    const scale = useSharedValue(1);
+
+    const opacityStyle = useAnimatedStyle(() => ({
+        opacity: opacity.value,
+    }));
+    const scaleStyle = useAnimatedStyle(() => ({
+        transform: [{scale: scale.value}],
+    }));
+
     const hideHasBeenCalled = useRef(false);
 
     const hide = useCallback(() => {
@@ -30,51 +39,44 @@ const SplashScreenHider = (props) => {
         hideHasBeenCalled.current = true;
 
         BootSplash.hide().then(() => {
-            Animated.timing(scale, {
+            scale.value = withTiming(0, {
                 duration: 200,
                 easing: Easing.back(2),
-                toValue: 0,
-                useNativeDriver: true,
-            }).start();
-
-            Animated.timing(opacity, {
-                duration: 250,
-                easing: Easing.out(Easing.ease),
-                toValue: 0,
-                useNativeDriver: true,
-            }).start(() => {
-                onHide();
             });
+
+            opacity.value = withTiming(
+                0,
+                {
+                    duration: 250,
+                    easing: Easing.out(Easing.ease),
+                },
+                () => runOnJS(onHide)(),
+            );
         });
     }, [opacity, scale, onHide]);
 
     return (
-        <Animated.View
+        <Reanimated.View
             onLayout={hide}
             style={[
                 StyleSheet.absoluteFill,
                 styles.splashScreenHider,
+                opacityStyle,
                 {
-                    opacity,
-
                     // Apply negative margins to center the logo on window (instead of screen)
                     marginTop: -(StatusBar.currentHeight || 0),
                     marginBottom: -(BootSplash.navigationBarHeight || 0),
                 },
             ]}
         >
-            <Animated.View
-                style={{
-                    transform: [{scale}],
-                }}
-            >
+            <Reanimated.View style={scaleStyle}>
                 <Logo
                     viewBox="0 0 100 100"
                     width={100}
                     height={100}
                 />
-            </Animated.View>
-        </Animated.View>
+            </Reanimated.View>
+        </Reanimated.View>
     );
 };
```

**Review by roryabraham — CHANGES_REQUESTED** [review]

File: src/components/SplashScreenHider/index.native.js
Comment (zoontek) [reply src/components/SplashScreenHider/index.native.js:1]: Done ✅

Comment (zoontek) [reply src/components/SplashScreenHider/index.native.js:1]: @roryabraham At first I used Animated because opacity / transform animations are natives, so there's no real impact. But I understand your point.

**Comment by roryabraham** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Review by roryabraham — APPROVED** [review]

**Comment by roryabraham** [comment]
Hmmm I had the ad-hoc build from this branch installed and noticed that the ReportActionContextMenu wasn't working. Let's monitor and see if the bug also exists on staging once this is deployed

File: src/Expensify.js
Comment (zoontek) [reply src/Expensify.js:174]: Ah sorry!

could be easily fixed with:

```ts
const shouldInit = isNavigationReady && (!isAuthenticated || props.isSidebarLoaded);

const shouldHideSplash = shouldInit && !isSplashHidden;
```

And use the first to wrap this, the second to wrap SplashScrenHider

File: src/components/SplashScreenHider/index.native.js
Comment (ahmedGaber93) [inline src/components/SplashScreenHider/index.native.js:60]: Coming from #57802, we notice that calling hide on onLayout cause hiding the native splash before the logo image is completely loaded, so we move calling it to onLoadEnd.

**Pull request merged**
